### ISSUE TITLE
mail notifications wiki points to a non-existing file

### ISSUE THREAD

**Issue by adrfantini** [netdata/netdata#3433]
On my ArchLinux server, the file `/usr/libexec/netdata/plugins.d/alarm-notify.sh`, which is referenced by the [email notifications wiki](https://github.com/firehol/netdata/wiki/email-notifications), does not exist.

Instead, one can use `/usr/lib/netdata/plugins.d/alarm-notify.sh`

**Comment by ktsaou** [comment]
I guess Arch has configured netdata to use `/usr/lib/netdata` instead of `/usr/libexec/netdata`.

@svenstaro is `/usr/libexec` obsolete?

**Comment by Ferroin** [comment]
@Ktsaou  Some older versions of the FHS did not support it, but the current version describes it as optional  (https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s07.html).  I think Arch decided to take the path of supporting as many versions of the FHS as possible in this case, though they should probably have that documented somewhere.

At a minimum though, there are a number of widely used software packages that do use it in the upstream configurations (GCC, Git, Postfix, Samba, RPM, CUPS, Amanda, BlueZ, Gnuplot, NetworkManager, and quite a few others).

Put simply, Netdata isn't in the wrong for using it, it's just that some distributions don't like to use it.

**Comment by ktsaou** [comment]
hm... would then make sense to use by default `/usr/lib` instead of `/usr/libexec` ? I mean if everyone is happy with `/usr/lib`, then why use `/usr/libexec`?

**Comment by Ferroin** [comment]
The separation on traditional UNIX systems was largely that you only put shared and static libraries in `/usr/lib`, while actual directly executable code went in `/usr/libexec`.  Strictly speaking, on the oldest UNIX systems, the stuff that's typically in `/usr/libexec` would have been in `/usr/bin` or `/usr/sbin`, not `/usr/lib`, because it's directly invokable code, not libraries for use by other programs.

For example, Amanda has a directory under `/usr/libexec` that contains all the individual programs and scripts that are run by the front-end components behind the scenes but aren't designed to be run directly by users.   The frontend commands for it will parse their options and then invoke the correct scripts and programs from `/usr/libexec/amanda` in the right order to achieve the desired result.  `/usr/lib/amanda` on the other hand contains the shared libraries that are part of Amanda.

Similarly, Postfix consists of a bunch of individual daemons that each do exactly one thing but aren't designed to be invoked directly when you want to start postfix.  Each of those daemon programs lives in `/usr/libexec/postfix`, and just like with Amanda, all the shared libraries that are part of Postfix are located in `/usr/lib/postfix`.

In the case of Netdata, I don't think it really matters much because there isn't any `libnetdata.so` to deal with.  The only thing I can think of that would have any impact whatsoever is that `/usr/lib` is _huge_ on most systems, so there might be marginally higher overhead doing the path lookup of `alarm-notify.sh` if it's under `/usr/lib`, but I doubt that it will be enough to matter in any practical case.  Either way though, I think you're likely to have some distros that choose to force it to be the other way (personally, I prefer use of `/usr/libexec` when it's done like Netdata, Postfix, and Amanda are currently doing it, but that's me).

**Comment by adrfantini** [comment]
I think this issue is still valid, but can't reopen it.
